core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -317,6 +377,14 @@ final class ShuffleBlockFetcherIterator(
         hostLocalBlockBytes += mergedBlockInfos.map(_.size).sum
       } else {
         remoteBlockBytes += blockInfos.map(_._2).sum
+        val inMemoryBlocks = blockInfos.filter(_._2 <= maxReqSizeShuffleToMem).map(_._2)
+        val maxInMemoryBlock = if (inMemoryBlocks.isEmpty) 0L else inMemoryBlocks.max
+        if (PlatformDependent.directBufferPreferred() &&
+            PlatformDependent.maxDirectMemory() < maxInMemoryBlock) {
+          throw new SparkException(
+            s"Netty memory (${PlatformDependent.maxDirectMemory()} bytes) is too small to " +
+              s"serve the max in-memory shuffle block ($maxInMemoryBlock bytes).")
+        }
```

Comment:
This is a case of misconfiguration (`PlatformDependent.maxDirectMemory() < maxReqSizeShuffleToMem`).
Should we move this check to initialization of executor itself - given this was an explicit configuration which is incorrect ?

Reply:
I don't think so. Here, we are not checking the case of `PlatformDependent.maxDirectMemory() < maxReqSizeShuffleToMem`, but the case of `PlatformDependent.maxDirectMemory() < maxInMemoryBlock` (where `maxInMemoryBlock` could be less than `maxReqSizeShuffleToMem`). For example, we assume the block fetching could be successful if maxDirectMemory=500M,  maxReqSizeShuffleToMem=600M,  maxInMemoryBlock=400M.

Reply:
`PlatformDependent.maxDirectMemory() < maxReqSizeShuffleToMem` is a stricter check right, with `maxReqSizeShuffleToMem` being the upper bound of `maxInMemoryBlock ` ?
You are right, in some cases the application might squeeze past where `maxInMemoryBlock` happened to be smaller than `PlatformDependent.maxDirectMemory()` even though `maxReqSizeShuffleToMem` is larger due to partition sizes sizes - but that is still a misconfig IMO - a failure waiting to happen for future executions ?

Reply:
I see. I got your point. Make sense to me. (cc @attilapiros who pointed out the original idea around here)